src/vegasflow/stratified.py

```diff
@@ -0,0 +1,172 @@
+        hypercubes = [list(p) for p in product(hypercubes_one_dim, repeat=int(n_dim))]
+        hypercubes_tf=tf.convert_to_tensor(hypercubes)
+        self.hypercubes = hypercubes_tf
+
+        #Initialize n_ev
+        n_ev = tf.math.floor(n_events/tf.math.pow(self.n_strat,n_dim))
+        self.n_ev=tf.fill([1 ,len(hypercubes)], int(n_ev))
+        self.n_ev = tf.reshape(self.n_ev,[-1])
+
+        #correction of self.n_events due to samples per hypercube
+        self.n_events = int(n_ev*tf.math.pow(self.n_strat,n_dim))
+
+    def redistribute_samples(self,arr_var):
+        """Receives an array with the variance of the integrand in each hypercube
+        and recalculate the samples per hypercube according to VEGAS+ algorithm"""
+        
+        damped_arr_var = tf.pow(arr_var,BETA)
+        new_n_ev = tf.maximum(2,damped_arr_var * self.n_events/tf.reduce_sum(damped_arr_var))
+        self.n_ev = tf.math.floor(new_n_ev)
+    
+        
+    def _run_event(self, integrand, ncalls=None):
+
+        if ncalls is None:
+            n_events = self.n_events
+        else:
+            n_events = ncalls
+
+        # Generate all random numbers 
+        rnds = tf.random.uniform((n_events, self.n_dim), minval=0, maxval=1, dtype=DTYPE)
+         
+        # Pass random numbers in hypercubes
+        x, w, segm = generate_random_array(rnds,self.n_strat,self.n_ev,self.hypercubes)
+        
+        # compute integrand
+        xjac = w
+        tmp = integrand(x, n_dim=self.n_dim, weight=xjac) * xjac
+        tmp2 = tf.square(tmp)
+
+        # tensor containing resummed component for each hypercubes
+        ress = tf.math.segment_sum(tmp,segm)
+        ress2 = tf.math.segment_sum(tmp2,segm)
+
+        #if adaptive save variance of each hypercube
+        arr_var = None
+        if self.adaptive:
+            hypercube_volume = tf.cast(tf.math.pow(1/self.n_strat, self.n_dim),DTYPE)
+            arr_var = ress2 * tf.cast(self.n_ev,DTYPE)* tf.square(hypercube_volume) - tf.square(ress*hypercube_volume)
+        
+        return ress, ress2, arr_var
+
+
+    def _run_iteration(self):
+
+        ress, raw_ress2, arr_var = self.run_event()
+
+        # compute variance for each hypercube
+        ress2 = raw_ress2 * tf.cast(self.n_ev,DTYPE)
+        err_tmp2s = (ress2 - tf.square(ress))/(tf.cast(self.n_ev,DTYPE)-fone)
+        sigmas2 = tf.maximum(err_tmp2s,fzero)
+
+        
+        res = tf.reduce_sum(ress) / tf.cast(tf.math.pow(self.n_strat,self.n_dim),DTYPE)
+        sigma2 = tf.reduce_sum(sigmas2)/tf.cast(tf.math.pow(self.n_strat,self.n_dim),DTYPE)/self.n_events    
+        sigma = tf.sqrt(sigma2)
+
+        # If adaptive is active redistributes samples
+        if self.adaptive:
+            self.redistribute_samples(arr_var)
+
+        return res, sigma
+
+
+def plain_wrapper(*args):
+    """ Wrapper around PlainFlow """
+    return wrapper(StratifiedFlow, *args)
+
+
+
+
+@tf.function
+def symgauss(xarr, n_dim=None, **kwargs):
```

Comment:
Move this example to tests `src/vegasflow/tests/test_algs.py`.
Implement this in examples.